gremlin-go/driver/traversal.go

```diff
@@ -120,3 +119,248 @@ func (t *Traversal) getResults() (ResultSet, error) {
+	Shuffle Order = "shuffle"
+	Asc     Order = "asc"
+	Desc    Order = "desc"
+)
+
+type Pick string
+
+const (
+	Any  Pick = "any"
+	None Pick = "none"
+)
+
+type Pop string
+
+const (
+	First Pop = "first"
+	Last  Pop = "last"
+	All   Pop = "all"
+	Mixed Pop = "mixed"
+)
+
+type Scope string
+
+const (
+	Global Scope = "global"
+	Local  Scope = "local"
+)
+
+type T string
+
+const (
+	Id    T = "id"
+	Label T = "label"
+	Id_   T = "id_"
+	Key   T = "key"
+	Value T = "value"
+)
+
+type Operator string
+
+const (
+	Sum     Operator = "sum"
+	Minus   Operator = "minus"
+	Mult    Operator = "mult"
+	Div     Operator = "div"
+	Min     Operator = "min"
+	Max     Operator = "max"
+	Assign  Operator = "assign"
+	And     Operator = "and"
+	Or      Operator = "or"
+	AddAll  Operator = "addAll"
+	SumLong Operator = "sumLong"
+)
+
+type p struct {
+	operator string
+	values   []interface{}
+}
+
+// Predicate interface
+type Predicate interface {
+	// Between Predicate to determine if value is within (inclusive) the range of two specified values.
+	Between(args ...interface{}) Predicate
+	// Eq Predicate to determine if equal to.
+	Eq(args ...interface{}) Predicate
+	// Gt Predicate to determine if greater than.
+	Gt(args ...interface{}) Predicate
+	// Gte Predicate to determine if greater than or equal to.
+	Gte(args ...interface{}) Predicate
+	// Inside Predicate to determine if value is within range of specified values (exclusive).
+	Inside(args ...interface{}) Predicate
+	// Lt Predicate to determine if less than.
+	Lt(args ...interface{}) Predicate
+	// Lte Predicate to determine if less than or equal to.
+	Lte(args ...interface{}) Predicate
+	// Neq Predicate to determine if not equal to.
+	Neq(args ...interface{}) Predicate
+	// Not Predicate gives the opposite of the specified Predicate.
+	Not(args ...interface{}) Predicate
+	// Outside Predicate to determine of value is not within range of specified values (exclusive).
+	Outside(args ...interface{}) Predicate
+	// Test Evalutes Predicate on given argument.
+	Test(args ...interface{}) Predicate
+	// Within Predicate determines if value is within given list of values.
+	Within(args ...interface{}) Predicate
+	// Without Predicate determines if value is not within the specified
+	Without(args ...interface{}) Predicate
+	// And Predicate returns a Predicate composed of two predicates (logical AND of them).
+	And(args ...interface{}) Predicate
+	// Or Predicate returns a Predicate composed of two predicates (logical OR of them).
+	Or(args ...interface{}) Predicate
+}
+
+var P *p = &p{}
+
+func newP(operator string, values []interface{}) Predicate {
+	return &p{operator: operator, values: values}
+}
+
+func newPWithP(operator string, pp p, values []interface{}) Predicate {
+	pSlice := make([]interface{}, len(values)+1)
+	copy(pSlice, values)
+	pSlice[len(pSlice)-1] = pp
+	return &p{operator: operator, values: pSlice}
+}
+
+func (_ *p) Between(args ...interface{}) Predicate {
+	return newP("between", args)
```

Comment:
Note for all of these args they aren't expanded. I'm not sure if this is intentional? ```args...```